src/Symfony/Component/HttpKernel/Tests/EventListener/RouterListenerTest.php

```diff
@@ -217,4 +221,57 @@ public function testRequestWithBadHost()
         $listener = new RouterListener($requestMatcher, $this->requestStack, new RequestContext());
         $listener->onKernelRequest($event);
     }
+
+    public function testResourceNotFoundException()
+    {
+        $this->expectException(NotFoundHttpException::class);
+        $this->expectExceptionMessage('No route found for "GET https://www.symfony.com/path" (from "https://www.google.com")');
+
+        $context = new RequestContext();
+
+        $urlMatcher = $this->createMock(UrlMatcherInterface::class);
+
+        $urlMatcher->expects($this->any())
+            ->method('getContext')
+            ->willReturn($context);
+
+        $urlMatcher->expects($this->any())
+            ->method('match')
+            ->willThrowException(new ResourceNotFoundException());
+
+        $kernel = $this->createMock(HttpKernelInterface::class);
+        $request = Request::create('https://www.symfony.com/path');
+        $request->headers->set('referer', 'https://www.google.com');
+
+        $event = new RequestEvent($kernel, $request, HttpKernelInterface::MASTER_REQUEST);
+
+        $listener = new RouterListener($urlMatcher, $this->requestStack);
+        $listener->onKernelRequest($event);
+    }
+
+    public function testMethodNotAllowedException()
+    {
+        $this->expectException(MethodNotAllowedHttpException::class);
+        $this->expectExceptionMessage('No route found for "GET https://www.symfony.com/path": Method Not Allowed (Allow: POST)');
```

Comment:
```suggestion
        $this->expectExceptionMessage('No route found for "GET https://www.symfony.com/path": Method Not Allowed (allowed methods: POST)');
```
?

Reply:
not part of my change.

Reply:
it's HTTP talk for `405 Method Not Allowed`: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Allow

i'd keep it